### DISCUSSION TITLE
Inertia v2

### DISCUSSION THREAD

**Discussion by tylernathanreed** [laravel/nova-issues#6560] [Ideas & Feature Requests]
With [Inertia v2](https://v2.inertiajs.com/) now in beta, I wanted to open a thread to brainstorm how the new Inertia features could be implemented into Nova. I'm not looking for any resolution on this thread, just ideas regarding how the v2 features could be implemented into Nova. If the Nova maintainers have plans already, I'd love to hear them!

# Polling

Nova already supports [Polling](https://nova.laravel.com/docs/resources/#resource-polling), but it doesn't offer [throttling](https://v2.inertiajs.com/polling#throttling). Having the ability to throttle polling would be helpful, as it can reduce server load.

# Prefetching

Nova's URL and relation fields could make use of [prefetching](https://v2.inertiajs.com/prefetching). This would likely need to be opt-in behavior, so perhaps we could chain a `->prefetch()` method on these fields.

# Infinite Scrolling

I immediately see [Infinite Scrolling](https://v2.inertiajs.com/merging-props) as a new type of pagination. I definitely have some index pages that would benefit from this.

# Lazy Loading on Scroll

Inertia's [load when visible](https://v2.inertiajs.com/load-when-visible) would be super handy for pages that use a lot of metrics. Honestly, any component that asynchronously loads in should have some sort of `->lazyLoad()` helper that we could chain onto it.

**Comment by crynobone** [comment]
We will look into integrating Inertia 2 feature when we remove `inertiajs/inertia-laravel` v1. 

The current objective with Laravel Nova 5 is to allow applications to upgrade to Inertia.js v2 as soon as possible while still allowing applications locked to Inertia.js v1 to receive the latest code.
